spark/src/main/java/org/apache/iceberg/spark/actions/BaseSparkAction.java

```diff
@@ -152,14 +154,17 @@ protected Table newStaticTable(TableMetadata metadata, FileIO io) {
   protected Dataset<Row> buildValidDataFileDF(Table table) {
     JavaSparkContext context = new JavaSparkContext(spark.sparkContext());
     Broadcast<FileIO> ioBroadcast = context.broadcast(SparkUtil.serializableFileIO(table));
+    return loadAllManifestFileBean(table).filter((FilterFunction<ManifestFileBean>) manifest ->
+        manifest.content() == ManifestContent.DATA)
```

Comment:
should compare using `MnaifestContent.DATA.equals(...)`

Reply:
I guess == is also correct for enum